use reflection to parse object structure

freegate78 opened the pull request:
use javax.json to build json serialization
use gson and json simplifications to meet json coding rules

freegate78 committed:
use reflection to parse object structure
use javax.json to build json serialization
use gson and json simplifications to meet json coding rules

kataus commented on HW15-JSON/src/main/java/com/akproductions/myjson/MyJson.java:
```diff
@@ -0,0 +1,143 @@
+                    jsonObject.add(field.getName(), jsonObjectInner);
+
+                } else if (fieldClass.name().contains("COLLECTION")) {
+                    System.out.println("serialize  Collection " + fieldClassType);
```
Вот тут прям стало грустно. Мы читали блок по архитектуре, читали, а потом вот такой безумный if получаем. ДАвайте приведем в порядок структуру класса, сделаем больше переиспользования и более простую поддержку

freegate78 committed:
1) заменил if на factory в enum. сразу так хотел сделать, но не понимал куда засунуть полиморфизм, поэтому сначала сделал работающий прототип. потом более внимательно почитал enum и увидел что можно перегружать методы в нем.
2) заметил, что при создании объекта AnyObject из json был баг - дергался дефолтовый конструктор, который инициализировал все поля объекта, в итоге информация перезатиралась при деконвертировании. сделал lazy инициализацию полей в методе init() в builder-style
3) добавил обработку NPE exception, эта ошибка у меня не обрабатывалась еще с прошлых ДЗ, Сергей сначала в банкомате не заметил, а потом в департаменте заметил (https://github.com/freegate78/java-2020-03/pull/7). чтобы при неподдерживаемом типе поля - выводилось предупреждении о пропуске поля при сериализации
